packages/components/table-v2/src/composables/use-columns.ts

```diff
@@ -14,8 +14,15 @@ function useColumns(
   columns: Ref<AnyColumns>,
   fixed: Ref<boolean>
 ) {
+  const _columns = computed(() =>
+    unref(columns).map((column) => ({
+      ...column,
+      key: column.key ?? column.dataKey,
```

Comment:
```suggestion
      key: column.key ?? column.dataKey ?? index,
```

```ts
  key?: KeyType
  dataKey?: KeyType
```

Perhaps the `column.key ?? column.dataKey` cannot ensure that the `key` value always exists